maps/001_Happy.tmx

```diff
@@ -14,6 +16,11 @@
    <point/>
   </object>
   <object id="3" name="player" type="entity" x="174.667" y="156.333">
+    <properties>
+        <property name="base_creds" type="int" value="10"/>
+        <property name="gold" type="float" value="1"/>
+        <property name="silver" type="float" value="2"/>
+    </properties>
```

Comment:
Set these on the map level instead. Just make tmx.lua aware of Custom Properties.

```diff
diff --git a/lib/tmx.lua b/lib/tmx.lua
index a2820c9..dca5ce1 100644
--- a/lib/tmx.lua
+++ b/lib/tmx.lua
@@ -215,7 +215,8 @@ local parse = function(file_name, raw_file_content, map_directory, tileset_table
   -- This will be our map's final table:
   local parsed_map = {
     layers = {},
-    tilesets = {}
+    tilesets = {},
+    properties = {}
   }
   -- We get an array of elements, but only need the map element
   local parsed_xml = Xml.parse(raw_file_content)[2]
@@ -261,6 +262,13 @@ local parse = function(file_name, raw_file_content, map_directory, tileset_table
       table.insert(parsed_map.tilesets, tileset_tables[element.xarg.source])
     elseif element.label == 'tileset' then
       table.insert(parsed_map.tilesets, parse_tileset(element, error_suffix))
+    -- "Custom Properties" set at the map level
+    elseif element.label == 'property' then
+      if element.xarg.type == 'int' then
+        parsed_map.properties[element.xarg.name] = tonumber(element.xarg.value)
+      else
+        parsed_map.properties[element.xarg.name] = element.xarg.value
+      end
     end
   end
   assert(
```

Reply:
Thanks, that almost worked. I had to adjust it by adding a nested for loop that went through the properties lable table

```
for _, element in ipairs(parsed_xml) do
    if element.label == 'properties' then
        for _, arg in ipairs(element) do
            if arg.xarg.type == 'int' or arg.xarg.type == 'float' then
                parsed_map.properties[arg.xarg.name] = tonumber(arg.xarg.value)
            else
                parsed_map.properties[argt.xarg.name] = arg.xarg.value
            end
        end
```